src/functional/ffront/foast_to_itir.py

```diff
@@ -231,17 +233,37 @@ def _visit_shift(self, node: foast.Call, **kwargs) -> itir.FunCall:
                 return im.shift_(offset_name)(self.visit(node.func, **kwargs))
         raise FieldOperatorLoweringError("Unexpected shift arguments!")
 
-    def _visit_reduce(self, node: foast.Call, **kwargs) -> itir.FunCall:
+    def _make_reduction_expr(
+        self,
+        node: foast.Call,
+        op: Callable[[itir.Expr], itir.Expr],
+        init_expr: int | itir.Literal,
+        **kwargs,
+    ):
         lowering = InsideReductionLowering()
         expr = lowering.visit(node.args[0], **kwargs)
         params = list(lowering.lambda_params.items())
         return im.lift_(
             im.call_("reduce")(
-                im.lambda__("accum", *(param[0] for param in params))(im.plus_("accum", expr)),
-                0,
+                im.lambda__("acc", *(param[0] for param in params))(op(expr)),
+                init_expr,
             )
         )(*(param[1] for param in params))
 
+    def _visit_reduce(self, node: foast.Call, **kwargs) -> itir.FunCall:
+
+        return self._make_reduction_expr(node, lambda expr: im.plus_("acc", expr), 0, **kwargs)
+
+    def _visit_compare(self, node: foast.Call, **kwargs) -> itir.FunCall:
```

Comment:
```suggestion
    def _visit_max_over(self, node: foast.Call, **kwargs) -> itir.FunCall:
```
The aliasing seems unneccesary.

Reply:
I changed it to _visit_reduce_max_over because _visit_max_over is where this is called below